include/alp/structures.hpp

```diff
@@ -168,36 +168,55 @@ namespace alp {
 
 		struct General {
 			using inferred_structures = std::tuple< General >;
+		};
 
-			/**
-			 * @brief Static and runtime check to determine if a matrix view of structure TargetStructure
-			 * 		  and index mapping functions (IMFs) \a imf_l and \a imf_r can be defined over \a SrcStructure.
-			 * 
-			 * @tparam SrcStructure The underlying structure of the target view.
-			 * @param imf_l 		The IMF applied to the rows of the source matrix.
-			 * @param imf_r 		The IMF applied to the columns of the source matrix.
-
-			 * @return \a false if the function can determined that the new view may alter underlying assumptions 
-			 * 			associated with the source structure \a SrcStructure; \a true otherwise. 
-			 */
-
-			template< typename SrcStructure >
-			static bool isInstantiableFrom( alp::imf::IMF & imf_l, alp::imf::IMF & imf_r ) {
+		/**
+		 * @brief Static and runtime check to determine if a matrix view of structure TargetStructure
+		 * 		  and index mapping functions (IMFs) \a imf_l and \a imf_r can be defined over \a SourceStructure.
+		 *
+		 * @tparam SourceStructure The underlying structure of the source view.
+		 * @tparam TargetStructure The underlying structure of the target view.
+		 * @param imf_l            The IMF applied to the rows of the source matrix.
+		 * @param imf_r            The IMF applied to the columns of the source matrix.
+	
+		 * @return \a false if the function can determined that the new view may alter underlying assumptions
+		 * 			associated with the source structure \a SourceStructure; \a true otherwise.
+		 */
+		template< typename SourceStructure, typename TargetStructure >
+		struct isInstantiable {
+			template< typename ImfL, typename ImfR >
+			static bool check( const ImfL &imf_l, const ImfR &imf_r ) {
```

Comment:
Was this intended to be a compile-time check a la `static constexpr bool value = ...` ?

Reply:
@anyzelman I think this is the runtime counterpart. This is the kind of check we specify in `get_view` will be used only if it involves constant time resolution (e.g., using get_view to select a small (possibly variable-size) non-zero square from a triangular matrix) and only enabled in debug mode for more expensive cases.